core/xchain/entitlement/entitlement.go

```diff
@@ -62,6 +63,8 @@ func (t CheckOperationType) String() string {
 		return "ERC1155"
 	case ISENTITLED:
 		return "ISENTITLED"
+	case ETHBALANCE:
+		return "ETHBALANCE"
```

Comment:
If some of the supported chains don't use ETH as their native token should we rename `ETHBALANCE` to something like `NATIVE_COIN_BALANCE`?

Reply:
Yes, that's a good thing to highlight about decimals, but we don't actually convert the threshold according to the decimal value in this code, we compare threshold to the returned currency value directly, so I think that's more of a UI detail, to convert ETH to Wei, or whatever native currency we'd like to compare.

Reply:
For native token balance I think your comment makes sense, but let me check with Fei about use case as a final gate.